test/upgrade/prober/wathola/event/services.go

```diff
@@ -105,9 +121,27 @@ func (f *finishedStore) State() State {
 	return f.errors.state
 }
 
-func (f *finishedStore) Thrown() []string {
+func (f *finishedStore) DuplicatedThrown() []string {
+	return f.thrownHelper(TagDuplicated)
+}
+
+func (f *finishedStore) MissingThrown() []string {
+	return f.thrownHelper(TagMissing)
+}
+
+func (f *finishedStore) DefaultThrown() []string {
+	return f.thrownHelper(TagDefault)
+}
+
+func (f *finishedStore) thrownHelper(tag string) []string {
 	msgs := make([]string, 0)
-	for _, t := range f.errors.thrown {
+	thrownChannel := f.errors.defaultThrown
+	if tag == TagDuplicated {
+		thrownChannel = f.errors.duplicatedThrown
+	} else if tag == TagMissing {
+		thrownChannel = f.errors.missingThrown
+	}
```

Comment:
Let's avoid having those if statements. Instead of having a single method with `if` lurking inside, it's cleaner to have different methods, that do something and have common functionality for them in an additional private method.

Imagine a following code to prepare a fruit salad chunks:

```golang
func pealAndChop(fruit Fruit, tools Tools) []Chunks {
  if fruit.SkinIsHard {
    peelSkin(tools.Knife, fruit)
  }
  if fruit.hasSeeds {
    cutOfSeeds(tools.Knife, fruit)
  }
  sizing := sizeGroup(fruit.Size)
  var parts int
  if sizing == "small" {
    parts = 2 // like strawberry
  } else if size == "modest" {
    parts = 6 // like apple
  } else if size == "big" {
    parts = 20 // like watermelon
  }
  return chop(tool.Knife, fruit, parts)
}
```

It's quite complicated and hard to read. It's also susceptible to change. And changes will be there. Like for example what happens when we would like to do a salad with coconut? Or other type of nuts? Or a mango? Or we'd like a pomegranate (only seeds are interesting).

Code should be open to adding, closed to modification.

Instead, imagine we have a code:

```golang
func (f *Apple) pealAndChop(tools Tools) []Chunks {
  peelSkin(tools.Knife, f)
  cutOfSeeds(tools.Knife, f)
  return chop(tool.Knife, f, 6)
}
func (f *Strawberry) pealAndChop(tools Tools) []Chunks {
  return chop(tool.Knife, f, 2)
}
func (f *Coconut) pealAndChop(tools Tools) []Chunks {
  f.drain()
  peelSkin(tools.Hammer, f)
  return chop(tool.Knife, f, 32)
}
// [..]
```

Such code if open to adding new functions, but you would not be going back to change old ones.

Reply:
Originally there were only one `if`, but I agree method is more extensible.